### PR TITLE
More sanity byron random-specific benchmarks post restoration + JSON artifacts

### PR DESCRIPTION
**Pull request by KtorZ**
# Issue Number

#2032 

# Overview

- 71f646dc05ad7af08a0e55d23ebb40d242359583
  :round_pushpin: **more sanity benchmarks post-restoration**
  
- 771e590f47ae523c561ccc3e50b38d8035444857
  :round_pushpin: **revise benchmark logging to use structured logger instead of plain raw stdout   For consistency and also, better log outputs.**
  
- 2b439b797618d45679325ff8f65c9577e2a45b5c
  :round_pushpin: **export benchmark results in structured JSON files for easier post-processing**
  
- 257e9303272f12c014168ff0e1aaaec6f234e0b7
  :round_pushpin: **add 'importManyAddresses' to the benchmarks for comparison with single address import**

# Comments

Now also export JSON files as: 

```json
{
    "listAddressesTime": "2.037 s",
    "restoreTime": "964.3 s",
    "benchName": "50-percent-rnd",
    "readWalletTime": "938.2 ms",
    "utxoStatistics": {
        "distribution": {
            "10000000000000000": 0,
            "100000000000000": 4,
            "1000000000000": 1998,
            "10000000000": 5971,
            "100000000": 1130,
            "1000000": 805,
            "10000": 203,
            "100": 515,
            "1000000000": 3217,
            "100000000000": 9121,
            "10": 75,
            "45000000000000000": 0,
            "1000": 43,
            "10000000000000": 51,
            "1000000000000000": 5,
            "100000": 1661,
            "10000000": 710
        },
        "scale": "log10",
        "total": {
            "quantity": 5210508362190411,
            "unit": "lovelace"
        }
    },
    "importManyAddressesTime": "3.901 s",
    "importOneAddressTime": "1.706 s",
    "estimateFeesTime": "2.740 s"
}
```

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Random.hs [hunk 1]
```diff
@@ -94,6 +99,42 @@ import System.Random
 import qualified Data.Map as Map
 import qualified Data.Set as Set
 
+class RndStateLike s where
```
Comment (Anviking) [inline lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery/Random.hs:102]: 👌 (makes me wonder slightly again if records hadn't been nicer)

File: lib/shelley/bench/Restore.hs [hunk 2]
```diff
@@ -316,24 +319,160 @@ cardanoRestoreBench tr c socketFile = do
...
-    , estimatingFeesTime :: Time
-    , utxoStatistics :: UTxOStatistics
-    } deriving (Show, Generic)
+    , restoreTime :: Time
+    , readWalletTime :: Time
+    , listAddressesTime :: Time
+    , importOneAddressTime :: Time
+    , importManyAddressesTime :: Time
+    , estimateFeesTime :: Time
+    , utxoStatistics :: s
```
Comment (Anviking) [inline lib/shelley/bench/Restore.hs:349]: I don't see: why do you need to parametrise over `s`?

Comment (KtorZ) [reply lib/shelley/bench/Restore.hs:349]: Because I didn't want to define a Buildable instance for ApiUtxoStatistics.

**Review by Anviking — APPROVED** [review]
a few details I didn't look at but, lgtm, and seems like a good idea!

**Pull request merged**